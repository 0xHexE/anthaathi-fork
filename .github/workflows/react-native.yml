name: Checks

on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]
  workflow_call:

permissions:
  contents: read

jobs:
  test:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        app: [ anthaathi-commerce-mobile-client ]

    steps:
      - uses: actions/checkout@v3

      - name: Setup Node.js environment
        uses: actions/setup-node@v3.1.1
        with:
          node-version: 16.x
          cache: 'yarn'

      - name: Install yarn packages
        run: yarn install

      - name: Set up JDK 11
        uses: actions/setup-java@v3
        with:
          java-version: '11'
          distribution: 'temurin'

      - name: Lint Application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:lint

      - name: Test application
        uses: gradle/gradle-build-action@0d13054264b0bb894ded474f08ebb30921341cee
        with:
          arguments: :apps:${{ matrix.app }}:test
