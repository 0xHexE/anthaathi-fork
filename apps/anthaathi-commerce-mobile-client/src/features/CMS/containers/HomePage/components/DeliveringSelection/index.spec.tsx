import React from 'react';
import 'react-native';
import {fireEvent, render} from '@testing-library/react-native';
import {DeliveringSelection} from './index';
import {IntlProvider} from 'react-intl';

import locale from '../../../../../../compiled-locales/en-US.json';
import {ThemeProvider} from 'react-native-paper';

describe('DeliveringSection', () => {
  it('should render the item', function () {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveringSelection
            onPress={() => {}}
            country="India"
            location="Dubai"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp).toMatchSnapshot();
    expect(temp.queryByTestId('deliveringSelection')).toBeTruthy();
  });

  it('should when we call tap', function () {
    const onpress = jest.fn();

    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveringSelection
            onPress={onpress}
            country="India"
            location="Dubai"
          />
        </IntlProvider>
      </ThemeProvider>,
    );

    fireEvent.press(temp.queryByTestId('deliveringSelection')!);
    expect(onpress).toBeCalledTimes(1);
  });

  it('should have DeliveringSelection country, location name', () => {
    const temp = render(
      <ThemeProvider>
        <IntlProvider locale="en-US" messages={locale}>
          <DeliveringSelection country="India" location="Dubai" />
        </IntlProvider>
      </ThemeProvider>,
    );

    expect(temp.queryByText('Dubai - India')).toBeTruthy();
    expect(temp.queryByText('Delivering to')).toBeTruthy();
  });
});
