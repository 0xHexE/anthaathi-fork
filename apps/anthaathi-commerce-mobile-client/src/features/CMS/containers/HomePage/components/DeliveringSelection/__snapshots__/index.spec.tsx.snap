// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveringSection should render the item 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  testID="deliveringSelection"
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 12,
          "display": "flex",
          "flexDirection": "row",
          "paddingHorizontal": 6,
          "paddingVertical": 0,
        },
        Object {
          "backgroundColor": "rgba(234, 221, 255, 1)",
          "borderColor": "rgba(103, 80, 164, 1)",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "center",
          "width": 38,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          Array [
            Object {
              "color": "rgba(103, 80, 164, 1)",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "Material Design Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        󰍎
      </Text>
    </View>
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 12,
              "fontWeight": "500",
              "letterSpacing": 0.5,
              "lineHeight": 16,
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
      >
        Delivering to
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "rgba(28, 27, 31, 1)",
              "fontSize": 14,
              "fontWeight": "500",
              "letterSpacing": 0.1,
              "lineHeight": 20,
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "writingDirection": "ltr",
            },
            undefined,
          ]
        }
      >
        Dubai
         - 
        India
      </Text>
    </View>
    <View
      style={
        Object {
          "flexGrow": 1,
        }
      }
    />
    <View
      collapsable={false}
      style={
        Object {
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": undefined,
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderWidth": 0,
              "elevation": 0,
              "height": 40,
              "margin": 6,
              "overflow": "hidden",
              "width": 40,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": false,
              }
            }
            accessible={true}
            focusable={true}
            hitSlop={
              Object {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
              ]
            }
          >
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "rgba(103, 80, 164, 1)",
                    "fontSize": 24,
                  },
                  Array [
                    Object {
                      "lineHeight": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              󰅂
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
